Utils/Elasticsearch/raw/oracle2es.php

```diff
@@ -0,0 +1,102 @@
+    "extradimensions", "warpeded", "contactInteraction","contactinteraction",'SEESAW'),
+    "GammaJets" => array("photon", "diphoton"),
+    "Higgs" => array("WHiggs", "ZHiggs", "mH125", "Higgs", "VBF", "SMHiggs", "higgs", "mh125",
+    "zhiggs", "whiggs", "bsmhiggs", "chargedHiggs","BSMHiggs","smhiggs"),
+    "Minbias" => array("minBias", "minbias"),
+    "Multijet" => array("dijet", "multijet", "qcd"),
+    "Performance" => array("performance"),
+    "SingleParticle" => array("singleparticle"),
+    "SingleTop" => array("singleTop", "singletop"),
+    "SUSY" => array("SUSY", "pMSSM", "leptoSUSY", "RPV", "bino", "susy", "pmssm", "leptosusy", "rpv",'MSSM'),
+    "Triboson" => array("tripleGaugeCoupling", "triboson", "ZZW", "WWW", "triplegaugecoupling", "zzw", "www"),
+    "TTbar" => array("ttbar"),
+    "TTbarX" => array("ttw","ttz","ttv","ttvv","4top","ttW","ttZ","ttV","ttWW","ttVV"),
+    "Upgrade" => array("upgrad"),
+    "Wjets" => array("W", "w"),
+    "Zjets" => array("Z", "z"),
+  );
+  foreach ($PHYS_CATEGORIES_MAP as $pc => $pcm) {
+    unset($PHYS_CATEGORIES_MAP[$pc]);
+    $pc = strtolower($pc);
+    $PHYS_CATEGORIES_MAP[$pc] = array();
+    foreach ($pcm as $pctag) {
+      $PHYS_CATEGORIES_MAP[$pc][] = strtolower($pctag);
+    }
+  }
+  foreach ($tags as $tag) {
+    foreach ($PHYS_CATEGORIES_MAP as $pc=>$tctags) {
+      if (in_array(strtolower($tag), $tctags)) {
+        $result[] = $pc;
+      }
+    }
+  }
+  if (count($result) == 0 ) {
+    $result[] = 'unknown';
+  }
+  return $result;
+}
+
+if (isset($argv[1])) {
+  $h = fopen($argv[1], "r");
+} else {
+  $h = fopen('php://stdin', 'r');
+}
+
+if ($h) {
+  while (($line = fgets($h)) !== false) {
+    $row = json_decode(strtolower($line),true);
+    if (!isset($row['sub_campaing'])) {
+      $row['sub_campaing'] = $row['campaing'];
+    }
+    printf('{ "index" : {"_index":"raw_current-%s", "_type":"%s", "_id":"%d" } }'."\n", $row['step'], $row['status'], $row['task_id']);
+    $row['hashtag'] = explode(',', $row['hashtag_list']);
+    $row['physics_category'] = getPhysCat($row['hashtag']);
+    $row['extended_tags'] = $row['hashtag'];
+    $tn = explode('.', $row['taskname']);
+    if (isset($tn[2])) {
+      $row['physics_short_full'] = $tn[2];
+      $row['physics_short'] = explode('_', $row['physics_short_full']);
+    }
+    $row['prod_tags'] = explode('_', $tn[count($tn)-1]);
+    $additional_hashtags = array('campaing', 'sub_campaing', 'phys_group', 'prod_tags', 'physics_category');
+
+
+    foreach ($additional_hashtags as $addon) {
+      if (isset($row[$addon])) {
+        if (is_array($row[$addon])) {
+          foreach($row[$addon] as $ra) {
+            if ($ra != '') {
+              $row['extended_tags'][] = $ra;
+            }
+          }
+        } else {
+          if ($row[$addon] != '') {
+            $row['extended_tags'][] = $row[$addon];
+          }
+        }
+      }
+    }
+    foreach( $row['physics_short'] as $ps) {
```

Comment:
As we lay on the taskname nomenclature we can of course be sure that the second field of the taskname exists and thus this key `physics_short` will be defined, I just wanted to point out to the potential of the crush.
If we define this key within an `if` clause, we can never actualy be sure ;)

Reply:
fixed